pilot/pkg/model/push_status.go

```diff
@@ -342,20 +345,26 @@ func (ps *PushStatus) GetServiceAttributes(hostname Hostname) (*ServiceAttribute
 
 	// For k8s - it just parses the name and returns UID, while checking
 	// for existence
-	name, namespace, err := parseHostname(hostname)
-	if err != nil {
-		return &ServiceAttributes{
+
+	// TODO: lookup the service by hostname (using a hashmap), and use labels
+	// for namespace.
+
+	// TODO: handle custom suffix
+	if strings.HasSuffix(string(hostname), ".cluster.local") {
```

Comment:
This is not good. Customers use non-default cluster suffix (`cluster.local` is just a parameters in kube admin).